source/extensions/tracers/xray/tracer.h

```diff
@@ -1,19 +1,83 @@
 #pragma once
 
 #include <string>
+#include <utility>
+#include <vector>
 
 #include "envoy/common/time.h"
 #include "envoy/tracing/http_tracer.h"
 
 #include "extensions/tracers/xray/sampling_strategy.h"
+#include "extensions/tracers/xray/xray_configuration.h"
 
+#include "absl/container/flat_hash_map.h"
 #include "absl/strings/string_view.h"
 
 namespace Envoy {
 namespace Extensions {
 namespace Tracers {
 namespace XRay {
 
+static const char XRayTraceHeader[] = "x-amzn-trace-id";
+
+class Span : public Tracing::Span {
+public:
+  Span(TimeSource& time_source) : time_source_(time_source) {}
+  void setTraceId(absl::string_view trace_id) { trace_id_ = std::string(trace_id); };
+  const std::string& traceId() const { return trace_id_; }
+  Span* childSpan() const;
+  void finishSpan() override;
+  void setOperation(absl::string_view operation) override {
+    operation_name_ = std::string(operation);
+  }
+  void setName(absl::string_view name) { name_ = std::string(name); }
+  void setTag(absl::string_view name, absl::string_view value) override {
+    // TODO(marcomagdy): add the values to either the annotations map or the metadata map
```

Comment:
Why is this a `TODO` and not implemented now?

Reply:
Because 
1. I'm not yet 100% sure how X-Ray determines what is metadata and what is annotations.
1. I want to make incremental changes and I feel the PR is already getting too big